src/sparse_blas/backends/mkl_common/mkl_spmv.cxx

```diff
@@ -0,0 +1,279 @@
+        }
+        else if (A_view.type_view == matrix_descr::symmetric ||
+                 A_view.type_view == matrix_descr::hermitian) {
+            oneapi::mkl::sparse::symv(queue, A_view.uplo_view, host_alpha, backend_handle, x_buffer,
+                                      host_beta, y_buffer);
+        }
+        else {
+            oneapi::mkl::sparse::gemv(queue, opA, host_alpha, backend_handle, x_buffer, host_beta,
+                                      y_buffer);
+        }
+        // Dependencies are not used for buffers
+        return {};
+    }
+    else {
+        auto x_usm = x_handle->get_usm_ptr<T>();
+        auto y_usm = y_handle->get_usm_ptr<T>();
+        if (A_view.type_view == matrix_descr::triangular) {
+            return oneapi::mkl::sparse::trmv(queue, A_view.uplo_view, opA, A_view.diag_view,
+                                             host_alpha, backend_handle, x_usm, host_beta, y_usm,
+                                             dependencies);
+        }
+        else if (A_view.type_view == matrix_descr::symmetric ||
+                 A_view.type_view == matrix_descr::hermitian) {
+            return oneapi::mkl::sparse::symv(queue, A_view.uplo_view, host_alpha, backend_handle,
+                                             x_usm, host_beta, y_usm, dependencies);
+        }
+        else {
+            return oneapi::mkl::sparse::gemv(queue, opA, host_alpha, backend_handle, x_usm,
+                                             host_beta, y_usm, dependencies);
+        }
+    }
+}
+
+sycl::event spmv(sycl::queue &queue, oneapi::mkl::transpose opA, const void *alpha,
+                 oneapi::mkl::sparse::matrix_view A_view,
+                 oneapi::mkl::sparse::matrix_handle_t A_handle,
+                 oneapi::mkl::sparse::dense_vector_handle_t x_handle, const void *beta,
+                 oneapi::mkl::sparse::dense_vector_handle_t y_handle,
+                 oneapi::mkl::sparse::spmv_alg alg, oneapi::mkl::sparse::spmv_descr_t spmv_descr,
+                 const std::vector<sycl::event> &dependencies) {
+    bool is_alpha_host_accessible = detail::is_ptr_accessible_on_host(queue, alpha);
+    bool is_beta_host_accessible = detail::is_ptr_accessible_on_host(queue, beta);
+    check_valid_spmv(__func__, opA, A_view, A_handle, x_handle, y_handle, is_alpha_host_accessible,
+                     is_beta_host_accessible);
+
+    if (!spmv_descr->optimized_called) {
+        throw mkl::uninitialized(
+            "sparse_blas", __func__,
+            "spmv_optimize must be called with the same arguments before spmv.");
+    }
+    CHECK_DESCR_MATCH(spmv_descr, opA, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, A_view, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, A_handle, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, x_handle, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, y_handle, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, alg, "spmv_optimize");
```

Comment:
These need to be _copied_ to `spmv_optimize()` API since, as mentioned in my other comment, these would need to be set in the `spmv_buffer_size` API.

Reply:
Resolving as per my comment, https://github.com/oneapi-src/oneMKL/pull/500#discussion_r1739212240